src/components/clusterWizard/wizardTransition.tsx

```diff
@@ -0,0 +1,58 @@
+import React from 'react';
+import { Cluster } from '../../api';
+import ClusterWizardContext from './ClusterWizardContext';
+import { CLUSTER_STEPS_INDEXED, CLUSTER_STEPS_ORDER } from './constants';
+
+// TODO(mlibra): I prefer collocating all these canNext* functions for easier maintenance.
+// They should be independent on each other anyway, so other option is to place them
+// close to the corresponding Form component.
+type TransitionProps = { isValid?: boolean; isSubmitting?: boolean; cluster: Cluster };
+
+export const canNextClusterConfiguration = ({
+  isValid,
+  isSubmitting,
+}: // cluster,
+TransitionProps) => {
+  let uiValidation = true;
+  if (isValid !== undefined) {
+    uiValidation = isValid && !isSubmitting;
+  }
+
+  // TODO(mlibra): check backend validations
+  const backendValidation = true;
+
+  return uiValidation && backendValidation;
+};
+
+export const canNextNetwork = ({ isValid, isSubmitting }: TransitionProps) => {
+  let uiValidation = true;
+  if (isValid !== undefined) {
+    uiValidation = isValid && !isSubmitting;
+  }
+
+  // TODO(mlibra): check backend validations
+  const backendValidation = true;
+
+  return uiValidation && backendValidation;
+};
+
+const getNextClusterWizardStep = (step: string) =>
```

Comment:
I feel like with this module, we're trying to teach the wizard to know about the steps order and be able to go to next/prev step from anywhere. It is very similar to what PatternFly implementation does - it introduces a Wizard component which knows how to transition from step to step.

I think this is not really needed. We can follow simple rules:
- Each step knows what is it's next step
- It is possible to go to any previous step from each step
- It is only possible to the next step by clicking 'next' in a specific step

I think there is no need to generalize the transition functions (using TransitionProps) let's define canNext* functions on each step - these functions can be completely tailored to each individual step without needing to generalize it's parameters.

Reply:
We can have a 'utils' module where we keep the functions which are used by individual steps such as `getClusterNetworkStatus`, `getClusterBaremetalInventoryStatus`...